packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml

```
@@ -194,6 +194,114 @@ processors:
         ctx.related = ctx.related ?: [:];
         ctx.related.entity = entities;
 
+##
+# Extract origin, target, and related entity IDs
+##
+  - append:
+      field: actor.entity.id
+      value: "{{{json.protoPayload.authenticationInfo.principalSubject}}}"
+      if: ctx.json?.protoPayload?.authenticationInfo?.principalSubject != null
+  - append:
+      field: actor.entity.id
+      value: "{{{json.protoPayload.authenticationInfo.principalEmail}}}"
+      if: ctx.json?.protoPayload?.authenticationInfo?.principalEmail != null
+  - append:
+      field: target.entity.id
+      value: "{{{json.protoPayload.resourceName}}}"
+      if: ctx.json?.resource?.type != 'k8s_cluster' && ctx.json?.protoPayload?.resourceName != null
+  - append:
+      field: target.entity.id
+      value: "{{{json.resource.disk_id}}}"
+      if: ctx.json?.resouce?.disk_id != null
+  - script:
+      description: Appends any relevant entity to `related.entity` for all events
+      lang: painless
+      on_failure:
+        - set:
+            description: Adds error reason to the document
+            field: error.message
+            value: "{{{ _ingest.on_failure_message }}}"
+      source: |
+        void addValue(Set entities, def value) {
+          if (value != null && value != "") {
+            entities.add(value);
+          }
+        }
+
+        boolean isKubernetes = false;
+        if (ctx.json?.resource?.type != null) {
+          String typ = ctx.json.resource.type;
+          isKubernetes = (typ == "k8s_cluster" || typ == "gke_cluster" || typ == "kubernetes");
+        }
```

Fine as is, but here's an alternative:

```suggestion
        boolean isKubernetes = ["k8s_cluster", "gke_cluster", "kubernetes"].contains(ctx.json?.resource?.type);
```

---

That's a nice, concise way of setting the flag. I'll keep it in mind for the future.